### ISSUE TITLE
Parallel execution for verification tests and better logging

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#1074]
We want to be able to run verification tests in parallel, and also to improve the output.

- Add a ``--num-threads``, ``-t`` option to the cli, and pass this value to ``runner.run``.
- In runner.run, using concurrent.futures to run the tests in parallel 
- We want to get an indication of how quickly the tests are running, so add a tqdm progress monitor that gets updated each time a test completes. 
- There's a lot of prints that will break the output of this. So, we want to replace *all* print statements with logging.info/logging.debug. I think logging.info should show the test name, and then everything else (like command line arguments) should go to debug.
- We also want a command line argument to control how much logging we output. The "-v", "--verbose" argument is good for this. See [here](https://github.com/tskit-dev/tsinfer/blob/master/tsinfer/cli.py#L155) for an example of how to set the logging system up to correspond with the args.verbose value. This is how you add a verbose argument:

```python
 parser.add_argument(
        "-v", "--verbose", action="count", default=0, help="Increase the verbosity"
    )
```

So, when we run ``python3 verification.py``, by default verbose is equal to 0 and no logging messages are output. If we run ``python3 verification.py -v``, verbose is 1, and all the INFO messages are written out, and with ``-vv``, all the INFO and DEBUG messages are printed out.

Some of the tests have embedded tqdm counters, which will mess things up a bit. Let's not worry about these for now, and sort them out in a follow up.

**Comment by jeromekelleher** [comment]
Oh yeah, let's use ``daiquiri`` for the logging setup like we do in the lined example in tsinfer above. It gives nice coloured output for the logs.

**Comment by saunack** [comment]
Anything else that needs to be done?

**Comment by jeromekelleher** [comment]
Closed in #1078.
